eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -0,0 +1,242 @@
+            ->setDescription('Recreate search engine index')
+            ->addOption('iteration-count', 'c', InputOption::VALUE_OPTIONAL, 'Number of objects to be indexed in a single iteration', 20)
+            ->addOption('no-commit', null, InputOption::VALUE_NONE, 'Do not commit after each iteration')
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> indexes current configured database in configured search engine index.
+EOT
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+        $iterationCount = $input->getOption('iteration-count');
+        $noCommit = $input->getOption('no-commit');
+
+        $output->writeln('Creating search index for the engine: ' . get_parent_class($this->searchHandler));
+
+        if ($this->searchHandler instanceof ContentIndexing) {
+            $this->searchHandler->purgeIndex();
+            $this->createContentIndex($iterationCount, empty($noCommit));
+            // Make changes available for search
+            $this->searchHandler->commit();
+        } else {
+            $output->writeln('Search Handler ' . get_class($this->searchHandler) . ' does not support ContentIndexing. Nothing to do.');
+        }
+
+        $output->writeln('Finished creating search index for the engine: ' . get_parent_class($this->searchHandler));
+    }
+
+    /**
+     * Wrapper for indexing Content.
+     *
+     * @param int $iterationCount a number of object items to be indexed at once
+     * @param bool $commit commit search index after each iteration
+     */
+    private function createContentIndex($iterationCount, $commit)
+    {
+        $stmt = $this->getContentDbFieldsStmt(['count(id)']);
+        $totalCount = intval($stmt->fetchColumn());
+
+        $stmt = $this->getContentDbFieldsStmt(['id', 'current_version']);
+
+        $this->searchHandler->purgeIndex();
+
+        /** @var \Symfony\Component\Console\Helper\ProgressBar $progress */
+        $progress = new ProgressBar($this->output);
+        $progress->start($totalCount);
+        $i = 0;
+        do {
+            $contentObjects = [];
+            $locations = [];
+
+            for ($k = 0; $k <= $iterationCount; ++$k) {
+                if (!$row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+                    break;
+                }
+
+                try {
+                    $content = $this->persistenceHandler->contentHandler()->load(
+                        $row['id'],
+                        $row['current_version']
+                    );
+                    $contentObjects[] = $content;
+
+                    // Skip location indexing if search engine does not use it, or if content does not have locations
+                    if (!($this->searchHandler instanceof LocationIndexing) || empty($content->versionInfo->contentInfo->mainLocationId)) {
```

Comment:
Indeed we're fetching Locations while indexing Content on purpose. 
Locations are indexed if `mainLocationId` of a `Content` exists. 
So, if I'm to move this part after line 170 it would have to be [inside this loop](https://github.com/alongosz/ezpublish-kernel/blob/02563ce994088324df8b6901863a3020d7e17886/eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php#L165-L171) which also causes too much nesting. 

Could also store all `mainLocationId`s in an array and fetch Locations after the line 170. 

I made a backup of this command with separate indexing of Locations [here](https://github.com/alongosz/ezpublish-kernel/blob/96f5878c2e304703dfe7c667eb6c483285309ed6/eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php#L89) to compare.